PropertyChanged.Fody/MappingFinder.cs

```diff
@@ -26,6 +26,14 @@ public static IEnumerable<MemberMapping> GetMappings(TypeDefinition typeDefiniti
                                  FieldDefinition = fieldDefinition
                              };
         }
+
+        if (typeDefinition.BaseType != null)
+        {
+            foreach (var mapping in GetMappings(typeDefinition.BaseType.Resolve()))
```

Comment:
`TypeNode` represents a tree of types, so it would be better to reuse that (and probably add a `BaseType` property), than to rescan all base types each time. You'd need to add logic to `IsPropertyGetInstruction` though, but that would reduce the scope of the change, which would be safer.

I think that adding info from base types could cause subtle issues here, for instance reading private fields from a base class. The existing code basically assumes that the mappings only contain data for the current class.

Reply:
Hmm, actually on second thought, my suggested solution wouldn't work if the base type is in a different assembly, while your current implementation probably would.

Maybe we should implement some additional search logic if a referenced property is virtual, but I'm not yet sure about the best approach here.

Reply:
If I understand you correctly, you're concerned about about emitting ldfld instructions for non-accessible fields? I haven't come up with any code-to-be-woven where I think that would happen, so I may be offering solutions to the wrong problem, but here are some things I've considered:

1) A conditional `yield return` instead:
```c#
if (mapping.PropertyDefinition.GetMethod.IsVirtual)
{
    yield return mapping;
}
```
This reduces the number of base members in the `Mappings` list, but ignores your concern about private fields.

2) Clear any `MemberMapping` fields that contain private members of base types:
```c#
if (mapping.FieldDefinition.IsPrivate)
{
    mapping.FieldDefinition = null;
}
if (mapping.PropertyDefinition.GetMethod.IsPrivate)
{
    mapping.PropertyDefinition = null;
}
```
This addresses your concern more directly, but is not completely correct. An internal member is not accessible to a derived class in a different assembly, but that wouldn't be cleared by these checks.

3) Instead of checking `IsPrivate` above, perform a full accessibility check.
I'd prefer not to do this. The code I've started writing is already unpleasant, and I haven't gotten to nested classes and the InternalsVisibleToAttribute yet. If there is an `IsAccessibleFrom` method, that removes my concern, but I haven't found one.